workos/organizations.py

```diff
@@ -47,114 +50,30 @@ def list_organizations(
-            "order": order or "desc",
+            "order": order,
         }
 
-        if order is not None:
-            if isinstance(order, Order):
-                params["order"] = str(order.value)
-
-            elif order == "asc" or order == "desc":
-                params["order"] = order
-            else:
-                raise ValueError("Parameter order must be of enum type Order")
-
         response = self.request_helper.request(
             ORGANIZATIONS_PATH,
             method=REQUEST_METHOD_GET,
             params=params,
             token=workos.api_key,
         )
 
-        response["metadata"] = {
-            "params": params,
-            "method": Organizations.list_organizations,
-        }
-
-        if "default_limit" in locals():
-            if "metadata" in response and "params" in response["metadata"]:
-                response["metadata"]["params"]["default_limit"] = default_limit
-            else:
-                response["metadata"] = {"params": {"default_limit": default_limit}}
-
-        return response
-
-    def list_organizations_v2(
-        self,
-        domains=None,
-        limit=None,
-        before=None,
-        after=None,
-        order=None,
-    ):
-        """Retrieve a list of organizations that have connections configured within your WorkOS dashboard.
-
-        Kwargs:
-            domains (list): Filter organizations to only return those that are associated with the provided domains. (Optional)
-            limit (int): Maximum number of records to return. (Optional)
-            before (str): Pagination cursor to receive records before a provided Organization ID. (Optional)
-            after (str): Pagination cursor to receive records after a provided Organization ID. (Optional)
-            order (Order): Sort records in either ascending or descending order by created_at timestamp.
-
-        Returns:
-            dict: Organizations response from WorkOS.
-        """
-
-        if limit is None:
-            limit = RESPONSE_LIMIT
-            default_limit = True
-
-        params = {
-            "domains": domains,
-            "limit": limit,
-            "before": before,
-            "after": after,
-            "order": order or "desc",
-        }
-
-        if order is not None:
-            if isinstance(order, Order):
-                params["order"] = str(order.value)
-
-            elif order == "asc" or order == "desc":
-                params["order"] = order
-            else:
-                raise ValueError("Parameter order must be of enum type Order")
-
-        response = self.request_helper.request(
-            ORGANIZATIONS_PATH,
-            method=REQUEST_METHOD_GET,
-            params=params,
-            token=workos.api_key,
+        return WorkOsListResource[Organization](
+            list_method=self.list_organizations,
+            # TODO: Should we even bother with this validation?
+            list_args=ListArgs.model_validate(params),
```

Comment:
I feel like the call made Friday was that we wouldn't focus on inbound param validation. I'm okay with either, but we should probably be consistent.

Reply:
Will remove in a follow-up.